lm_eval/evaluator.py

```diff
@@ -133,21 +133,18 @@ def simple_evaluate(
             if task_obj is None:
                 continue
 
-        if predict_only:
-            log_samples = True
-            eval_logger.info(
-                f"Processing {task_name} in output-only mode. Metrics will not be calculated!"
-            )
-            # we have to change the class properties post-hoc. This is pretty hacky.
-            task_obj.override_metric(metric_name="bypass")
-
-        if (
-            task_obj.get_config("output_type") == "generate_until"
-            and gen_kwargs is not None
-        ):
-            task_obj.override_config(
-                key="generation_kwargs", value=gen_kwargs, update=True
-            )
+        if task_obj.get_config("output_type") == "generate_until":
```

Comment:
We should add a warning that  `--predict_only` will be ignored for non-generate_until tasks.

Reply:
It works for all. Should I restrict it to generate_until?

Reply:
misread the behavior, I think allowing it to work for all is more intuitive, that’s fine!